src/SFA.DAS.Tasks.Worker/MessageProcessors/SignedEmployerAgreementMessageProcessor.cs

```diff
@@ -16,22 +17,50 @@ public class SignedEmployerAgreementMessageProcessor : MessageProcessor<Agreemen
         private readonly ILog _logger;
         private readonly IMediator _mediator;
 
-        public SignedEmployerAgreementMessageProcessor(IMessageSubscriberFactory subscriberFactory, ILog logger, IMediator mediator) : base(subscriberFactory, logger)
+        public SignedEmployerAgreementMessageProcessor(IMessageSubscriberFactory subscriberFactory, ILog logger, IMediator mediator) 
+            : base(subscriberFactory, logger)
         {
             _logger = logger;
             _mediator = mediator;
         }
         
         protected override async Task ProcessMessage(AgreementSignedMessage message)
```

Comment:
Can the internals of this method be refactored to 2 separate methods?  RemoveAgreementToSignTask and CreateAddApprenticesTask or something along those lines.

Reply:
sure, i'll do that now